web/src/components/storage/ProposalPage.test.jsx

```diff
@@ -76,32 +91,84 @@ const vdb = {
   size: 1e+6
 };
 
-const storageMock = {
-  probe: jest.fn().mockResolvedValue(0),
-  proposal: {
-    getAvailableDevices: jest.fn().mockResolvedValue([vda, vdb]),
-    getEncryptionMethods: jest.fn().mockResolvedValue([]),
-    getProductMountPoints: jest.fn().mockResolvedValue([]),
-    getResult: jest.fn().mockResolvedValue(undefined),
-    defaultVolume: jest.fn(mountPath => Promise.resolve({ mountPath })),
-    calculate: jest.fn().mockResolvedValue(0),
-  },
-  system: {
-    getDevices: jest.fn().mockResolvedValue([vda, vdb])
-  },
-  staging: {
-    getDevices: jest.fn().mockResolvedValue([vda])
-  },
-  getErrors: jest.fn().mockResolvedValue([]),
-  isDeprecated: jest.fn().mockResolvedValue(false),
-  onDeprecate: jest.fn(),
-  onStatusChange: jest.fn()
+/**
+ * @param {string} mountPath
+ * @returns {Volume}
+ */
+const volume = (mountPath) => {
+  return (
+    {
+      mountPath,
+      target: "DEFAULT",
+      fsType: "Btrfs",
+      minSize: 1024,
+      maxSize: 1024,
+      autoSize: false,
+      snapshots: false,
+      transactional: false,
+      outline: {
+        required: false,
+        fsTypes: ["Btrfs"],
+        supportAutoSize: false,
+        snapshotsConfigurable: false,
+        snapshotsAffectSizes: false,
+        sizeRelevantVolumes: [],
+        adjustByRam: false
+      }
+    }
+  );
 };
 
+/** @type {StorageClient} */
 let storage;
 
+/** @type {ProposalResult} */
+let proposalResult;
+
 beforeEach(() => {
-  storage = { ...storageMock, proposal: { ...storageMock.proposal } };
+  proposalResult = {
+    settings: {
+      target: "DISK",
+      targetPVDevices: [],
+      configureBoot: false,
+      bootDevice: "",
+      defaultBootDevice: "",
+      encryptionPassword: "",
+      encryptionMethod: "",
+      spacePolicy: "",
+      spaceActions: [],
+      volumes: [],
+      installationDevices: []
+    },
+    actions: []
+  };
+
+  storage = {
+    probe: jest.fn().mockResolvedValue(0),
+    // @ts-expect-error Some methods have to be private to avoid type complaint.
+    proposal: {
+      getAvailableDevices: jest.fn().mockResolvedValue([vda, vdb]),
+      getEncryptionMethods: jest.fn().mockResolvedValue([]),
+      getProductMountPoints: jest.fn().mockResolvedValue([]),
+      getResult: jest.fn().mockResolvedValue(proposalResult),
+      defaultVolume: jest.fn(mountPath => Promise.resolve(volume(mountPath))),
+      calculate: jest.fn().mockResolvedValue(0),
+    },
+    // @ts-expect-error Some methods have to be private to avoid type complaint.
+    system: {
+      getDevices: jest.fn().mockResolvedValue([vda, vdb])
+    },
+    // @ts-expect-error Some methods have to be private to avoid type complaint.
+    staging: {
+      getDevices: jest.fn().mockResolvedValue([vda])
+    },
+    getErrors: jest.fn().mockResolvedValue([]),
+    isDeprecated: jest.fn().mockResolvedValue(false),
+    onDeprecate: jest.fn(),
+    onStatusChange: jest.fn()
+  };
+
+  // @ts-expect-error
```

Comment:
NP: adds a hint why a ts error is expected

Reply:
Thanks a lot for adding it. I found a way to fix it by not expecting an error, see https://gist.github.com/dgdavid/99916d7571697b9b3b733d71e6191fd3